src/Compilers/CSharp/Test/Emit/CodeGen/PatternTests.cs

```diff
@@ -3236,39 +3233,37 @@ static bool M(object o)
                 var compVerifier = CompileAndVerify(compilation, expectedOutput: expectedOutput);
                 if (options.OptimizationLevel == OptimizationLevel.Debug)
                 {
-                    compVerifier.VerifyIL("Class1.M",
-@"{
-  // Code size       39 (0x27)
-  .maxstack  2
-  .locals init (bool V_0,
-                int V_1,
-                bool V_2,
-                bool V_3)
-  IL_0000:  nop
-  IL_0001:  ldarg.0
-  IL_0002:  isinst     ""int""
-  IL_0007:  brfalse.s  IL_001b
-  IL_0009:  ldarg.0
-  IL_000a:  unbox.any  ""int""
-  IL_000f:  stloc.1
-  IL_0010:  ldloc.1
-  IL_0011:  ldc.i4.s   42
-  IL_0013:  beq.s      IL_0017
-  IL_0015:  br.s       IL_001b
-  IL_0017:  ldc.i4.1
-  IL_0018:  stloc.0
-  IL_0019:  br.s       IL_001f
-  IL_001b:  ldc.i4.0
-  IL_001c:  stloc.0
-  IL_001d:  br.s       IL_001f
-  IL_001f:  ldloc.0
-  IL_0020:  stloc.2
-  IL_0021:  ldloc.2
-  IL_0022:  stloc.3
-  IL_0023:  br.s       IL_0025
-  IL_0025:  ldloc.3
-  IL_0026:  ret
-}");
+                    compVerifier.VerifyIL("Class1.M", @"
+{
+      // Code size       37 (0x25)
+      .maxstack  2
+      .locals init (bool V_0,
+                    int V_1,
+                    bool V_2)
```

Comment:
How did this IL end up changing? It doesn't seem related to the changes in the implementation.

Reply:
See the change in SpillSequenceSpiller.  We no longer make an extra copy through a local of the value of a spill sequence.

---
In reply to: [311192368](https://github.com/dotnet/roslyn/pull/37711#discussion_r311192368) [](ancestors = 311192368)